community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -157,6 +158,26 @@
      */
     Type transactionType();
 
+    /**
+     * Return transaction id if it was already assigned or {@link TransactionData#UNASSIGNED_TRANSACTION_ID} otherwise.
+     * Transaction id is assigned as soon as transaction committed.
+     * @return transaction id.
+     */
+    default long getTransactionId()
```

Comment:
Default implementation was introduced because this interface is part of `org.neo4j.kernel.api` package with at least misleading name (is it public or not?) and since it's a point release - playing safe. 
If it's our internal interface we do not need default implementation at all.